#! /bin/sh -e

# DP: adjustments to XISDNLoad Xresources.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p1 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p1 < $0
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

--- isdnutils-3.3.orig/xisdnload/XISDNLoad.ad
+++ isdnutils-3.3/xisdnload/XISDNLoad.ad
@@ -1,15 +1,16 @@
 XISDNLoad.geometry:		250x100
 *Label*Justify:			left
-*Label*Label:			%7s %3d:%02d %4.0fcps %ldkB - %-100s
+! *Label*Label:			%7s %3d:%02d %4.0fcps %ldkB - %-100s
+*Label*Label:			%7s %3d:%02d %4.0fcps %ldkB
 ! *minScale:			8
-*minScale:			1
+*minScale:			2
 *JumpScroll:			1
 *update:			1
 *internalBorderWidth:		0
 *showGrip:			FALSE
 *onlineColor:			red
 *tryingColor:			yellow
-*activeColor:			green
-! *activate:			sudo route add default dev isdn2 &
-! *deactivate:			sudo route del default ; root isdnctrl hangup isdn2 &
+*activeColor:			lightgreen
+*activate:			/etc/isdn/xisdnload-netup &
+*deactivate:			/etc/isdn/xisdnload-netdown &
 *number:			.*
